app/src/pages/review.tsx

```diff
@@ -1,180 +1,151 @@
-}
+import {
+  EligibilityScreenerBody,
+  EligibilityScreenerResponse,
+} from '@pages/api/eligibility-screener'
 
-export const formatEligibilityResponses = (
-  session: DefaultState
-): ReviewElementProps[] => {
-  return [
-    {
-      labelKey: 'Eligibility.residential',
-      children:
-        (session?.eligibility?.residential && (
-          <Trans i18nKey={session?.eligibility?.residential} />
-        )) ||
-        null,
-    },
-    {
-      labelKey: 'Eligibility.categorical',
-      children: formatCategoricalOrAdjunctive(categoryKeys, session),
-    },
-    {
-      labelKey: 'Eligibility.before',
-      children:
-        (session?.eligibility?.before && (
-          <Trans i18nKey={session?.eligibility?.before.replace(/[2]/g, '')} />
-        )) ||
-        null,
-    },
-    {
-      labelKey: 'Eligibility.programs',
-      children: formatCategoricalOrAdjunctive(programKeys, session),
-    },
-  ]
-}
+import type { EditablePage } from '@src/types'
+import { initialSessionData } from '@utils/sessionData'
 
-export const formatClinicResponses = (
-  session: DefaultState
-): ReviewElementProps[] => {
-  return [
-    {
-      labelKey: 'Review.clinicSelected',
-      children:
-        (session?.clinic && (
-          <ClinicInfo
-            name={session?.clinic.clinic}
-            streetAddress={session?.clinic.clinicAddress}
-            phone={session?.clinic.clinicTelephone}
-            isFormElement={false}
-          />
-        )) ||
-        null,
-    },
-  ]
+interface ReviewProps extends EditablePage {
+  baseUrl: string
 }
 
-export const formatContactResponses = (
-  session: DefaultState
-): ReviewElementProps[] => {
-  const contactResponses: ReviewElementProps[] = []
-  contactKeys.forEach((key: string) => {
-    contactResponses.push({
-      labelKey: `Contact.${key}`,
-      children:
-        (session?.contact[key as Contact] && (
-          <Trans i18nKey={session?.contact[key as Contact]} />
-        )) ||
-        null,
-    })
-  })
-  return contactResponses
-}
+const Review: NextPage<ReviewProps> = (props: ReviewProps) => {
+  const {
+    session,
+    setSession,
+    backRoute = '',
+    forwardRoute = '',
+    baseUrl,
+  } = props
+
+  // Using form to store all of the data in a component state
+  // resolves all hydration issues.
+  const [form, setForm] = useState(initialSessionData)
+  useEffect(() => {
+    setForm(session)
+  }, [session])
+
+  const [errorMessage, setErrorMessage] = useState('')
+
+  const { t } = useTranslation('common')
+
+  const router = useRouter()
 
-const Review: NextPage = () => {
-  const { session } = useAppContext()
+  const buildEligibilityArrays = (data: string[]) => {
+    return data.map((category) => t(`Eligibility.${category}`))
+  }
+
+  // Note: All router.push() calls have linting disabled on them.
+  // See https://nextjs.org/docs/api-reference/next/router#potential-solutions
+  const handleClick = (e: MouseEvent<HTMLElement>) => {
```

Comment:
Notes for reviewers: This `handleClick` is very complicated. It is written to make the `fetch()` call in such a way as not to return a dangling promise. If there are no errors with the api call, the user gets routed to `/confirmation` via [next.js next/router](https://nextjs.org/docs/api-reference/next/router). If there are errors, then we reload the current page and display an error.

Reply:
wanted to say I read this over - not sure I fully grok the challenges you're solving for so not adding much wisdom here - wanted to at least respond though so you know I did read this over!